pyproject.toml

```diff
@@ -18,11 +18,16 @@ dependencies = [
     "apache-airflow >=2.0",
     "apache-airflow-providers-postgres",
     "apache-airflow-providers-snowflake",
+    "apache-airflow-providers-google",
+    "python-frontmatter",
     "pandas >=1.3.4",
     "s3fs",
     "snowflake-sqlalchemy ==1.2.0",
     "snowflake-connector-python[pandas]",
     "smart-open[all]>=5.2.1",
+    "SQLAlchemy == 1.4.28",
+    "snowflake-sqlalchemy",
+    "sqlalchemy-bigquery"
```

Comment:
@utkarsharma2 @dimberman I'd love to get the view of both of you on our approach towards Astro's dependencies

I just created isolated virtualenvs for both `main` and `SQLAlchemy-demo` branches by running:
```
pip install .[dependencies]
```

Some of the differences were that in `SQLAlchemy-demo` we downgraded a few packages:
- `apache-airflow==2.2.3` -> `apache-airflow==2.0.1`
- `apache-airflow-providers-postgres==2.4.0` -> `apache-airflow-providers-postgres==1.0.2 `
- `apache-airflow-providers-postgres==2.4.0` -> `apache-airflow-providers-snowflake==1.3.0`

Was this downgrade be intentional? Could this affect Astro's behaviour/adoption in any sense?

 Also, in `main`, `pip freeze` lists 145 dependencies, and the branch   `SQLAlchemy-demo` has 192. Are we happy for Astro to have this amount of dependencies, or do we value fewer dependencies? There are successful open source projects which followed both paths of lots of dependencies / minimal dependencies (e.g. Plone and Django) - it would be great to understand where we stand.